common/src/main/java/io/druid/metadata/DefaultPasswordProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2015 - Yahoo! Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.metadata;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.ISE;
+
+
+public class DefaultPasswordProvider implements PasswordProvider
+{
+
+  private final static String PASSWORD_KEY = "password";
+
+  private final String password;
+
+  @JsonCreator
+  public DefaultPasswordProvider(Object spec)
+  {
+    password = convertValue(spec);
+  }
+
+  private static String convertValue(Object spec) {
+    if(spec instanceof String) {
+      return spec.toString();
+    } else if(spec instanceof Map && ((Map)spec).containsKey(PASSWORD_KEY)) {
```

Comment:
@cheddar We shouldn't have to do the Map<> parsing if we let jackson handle the serde directly. All we need to do is change the defaultImpl to a separate class that has a JsonCreator method with a single string parameter (to support simple string), and leave everything else to use property annotations in the constructor (to handle the object style parsing)